msystems/services.py

```diff
@@ -121,3 +135,26 @@ def _add_new_user_policyholders(self, user: User, policyholders: List[PolicyHold
         for ph in policyholders:
             if ph not in current_policyholders:
                 PolicyHolderUser(user=user, policy_holder=ph).save(username=user.username)
+
+    def _update_roles(self, i_user, imis_role_ids):
+        self._remove_previous_user_roles(i_user)
+        self._connect_role_with_user(i_user, imis_role_ids)
+
+    def _connect_role_with_user(self, i_user, imis_role_ids):
+        for imis_role_id in imis_role_ids:
+            role = Role.objects.filter(is_system=imis_role_id).first()
+            UserRole.objects.create(user=i_user, role=role)
+
+    def _remove_previous_user_roles(self, i_user):
+        roles = UserRole.objects.filter(user=i_user)
+        if roles.exists():
+            for role in roles:
+                role.delete_history()
```

Comment:
Please make sure you only remove roles the user lost or add roles the user didn't already have, because rewriting user roles every time he logs in will waste database space with useless history

Reply:
done